onnxruntime/python/tools/quantization/E2E_example_model/e2e_user_example.py

```diff
@@ -0,0 +1,81 @@
+import os
+import sys
+import numpy as np
+import re
+import abc
+import subprocess
+import json
+from PIL import Image
+
+import onnx
+import onnxruntime
+from onnx import helper, TensorProto, numpy_helper
+from quantize import quantize, QuantizationMode
+from calibrate import calibrate
+from calibrate import CalibrationDataReader
+
+
+class ResNet50DataReader(CalibrationDataReader):
+    def __init__(self,calibration_image_folder,augmented_model_path='augmented_model.onnx'): 
+        self.image_folder = calibration_image_folder
+        self.augmented_model_path = augmented_model_path
+        self.preprocess_flag = True
+        self.enum_data_dicts = []
+        self.datasize = 0
+
+    def get_next(self):
+        if self.preprocess_flag:
+            self.preprocess_flag = False
+            session = onnxruntime.InferenceSession(self.augmented_model_path, None)
+            (_,height,width,_) = session.get_inputs()[0].shape
+            nhwc_data_list = preprocess_func(self.image_folder,height,width,size_limit = 0)
```

Comment:
>preprocess_func [](start = 29, length = 15)

is it legal to use a function before definition in python?

Reply:
Yes, I was wondering about this too.  But I have tested and run it and it works.

Reply:
> > preprocess_func [](start = 29, length = 15)
> 
> is it legal to use a function before definition in python?

There is no such thing in python like forward declaration.